### PR TITLE
Redis client [issue #143]

### PR DESCRIPTION
**Pull request by s-gruneberg**
Draft of a redis client. I have been running it locally using docker. 

/assign @xuanyang-cn

### CODE REVIEW

**Comment by liliu-z** [comment]
Hi @s-gruneberg Thanks for contribution!! BTW, can I ask is this for Redis Cloud or Redis opensource?

**Comment by s-gruneberg** [comment]
@liliu-z this is for redis open source, it is similar to the pgVector implementation where it takes a connection string. To my knowledge it should work with redis cloud if the connection to the cloud database is set up. It might take some slight tweaking.

**Comment by s-gruneberg** [comment]
I realized that my filter search is not working properly, closing this PR until I fix it.

**Comment by s-gruneberg** [comment]
> @s-gruneberg any update on this?

Yes, all unit tests are passing and filter search is working. As far as I can tell it's good to go. 
Let me know if you have any questions or need me to modify anything / fix errors!

**Comment by liliu-z** [comment]
Sure, let test this a little bit and get back to you! Plz know that this may take some time due to the lack of human power.

File: vectordb_bench/backend/clients/redis/config.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+from pydantic import SecretStr
+from ..api import DBConfig
+
+class RedisConfig(DBConfig):
+    name: str = "" 
+    password: str = SecretStr
+    host: str = ""
```
Comment (XuanYang-cn) [inline vectordb_bench/backend/clients/redis/config.py:7]: Please refer to https://github.com/zilliztech/VectorDBBench/blob/675b48f0d2b5561886af17dd897d789bd5e76f5b/vectordb_bench/backend/clients/qdrant_cloud/config.py#L8

SecretStr is a type not a value. Please change host to SecretStr type too. These are both sensitive info.

File: vectordb_bench/backend/clients/redis/config.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+from pydantic import SecretStr
+from ..api import DBConfig
+
+class RedisConfig(DBConfig):
+    name: str = ""
```
Comment (XuanYang-cn) [inline vectordb_bench/backend/clients/redis/config.py:5]: Same as Chorma, I don't see any usage of param name, please remove it if it's not essentail for connection with the server.

**Review by XuanYang-cn — CHANGES_REQUESTED** [review]

**Comment by XuanYang-cn** [comment]
It would be better if you could squish all of these changes together into one single commit. This will make it easier to track changes and maintain a more organized version history

**Comment by XuanYang-cn** [comment]
LGTM

@s-gruneberg Seems this PR has some conflicts with the chroma client one, could you resolve the conflicts?

**Review by XuanYang-cn — APPROVED** [review]

**Comment by XuanYang-cn** [comment]
@s-gruneberg Please squish the two commits together into one single commit. It cannot be auto rebased so unable to merge.

**Comment by s-gruneberg** [comment]
@XuanYang-cn 
I'm having some trouble squashing it: 
Github is flagging my squashed commit as having a merge conflict when as far as I can tell there is no conflict. Resolving conflict (on github) gives me an additional commit titled "Merge branch main..." with no differences from previous commit. 

I'm not sure how to fix this on my end because i'm going in circles. If I "resolve" merges on github it makes two commits. If I squash those commits to one (current state) github says there are merge conflicts.

Is it possible to squash on PR merge? If so I can 'resolve' the conflicts and end up with two commits again.

**Comment by s-gruneberg** [comment]
@XuanYang-cn 
Figured it out! Thanks for your patience!

**Pull request merged**